R/lazyframe-frame.R

```diff
@@ -2610,4 +2610,199 @@ lazyframe__sink_csv <- function(
+      name <- names(schema)[i]
+      dtype <- schema[[i]]
+
+      is_numeric <- dtype$is_numeric()
+      is_temporal <- !is_numeric && dtype$is_temporal()
+
+      # counts
+      count_exprs <- c(
+        pl$col(name)$count()$name$prefix("count:"),
+        pl$col(name)$null_count()$name$prefix("null_count:")
+      )
+
+      # mean
+      mean_expr <- if (is_temporal || is_numeric || inherits(dtype, "polars_dtype_boolean")) {
+        pl$col(name)$mean()
+      } else {
+        pl$lit(NA, dtype = dtype)
+      }
+
+      # standard deviation, min, max
+      expr_std <- if (is_numeric) {
+        pl$col(name)$std()
+      } else {
+        pl$lit(NA, dtype = dtype)
+      }
+      min_expr <- if (!skip_minmax(dtype)) {
+        pl$col(name)$min()
+      } else {
+        pl$lit(NA, dtype = dtype)
+      }
+      max_expr <- if (!skip_minmax(dtype)) {
+        pl$col(name)$max()
+      } else {
+        pl$lit(NA, dtype = dtype)
+      }
+
+      # percentiles
+      pct_exprs <- c()
+      for (index_quantile in seq_along(quantiles)) {
+        p <- quantiles[index_quantile]
+        pct_expr <- if (is_temporal) {
+          pl$col(name)$to_physical()$quantile(p, interpolation)$cast(dtype)
+        } else if (is_numeric) {
+          pl$col(name)$quantile(p, interpolation)
+        } else {
+          pl$lit(NA, dtype = dtype)
+        }
+        sort_cols <- c(sort_cols, name)
+        pct_exprs <- c(pct_exprs, pct_expr$alias(paste0(p, ":", name)))
+      }
+
+      if (
+        is_numeric ||
+          dtype$is_nested() ||
+          inherits(self, "polars_dtype_null") ||
+          inherits(dtype, "polars_dtype_boolean")
+      ) {
+        has_numeric_result <- c(has_numeric_result, name)
+      }
+
+      # add column expressions (in end-state 'metrics' list order)
+      metric_exprs <- c(
+        metric_exprs,
+        count_exprs,
+        mean_expr$alias(paste0("mean:", name)),
+        expr_std$alias(paste0("std:", name)),
+        min_expr$alias(paste0("min:", name)),
+        pct_exprs,
+        max_expr$alias(paste0("max:", name))
+      )
+    }
+
+    # calculate requested metrics in parallel, then collect the result
+    df_metrics <- if (length(sort_cols) > 0) {
+      self$with_columns(pl$col(!!!unique(sort_cols))$sort())
+    } else {
+      self
+    }
+
+    df_metrics <- df_metrics$select(!!!metric_exprs)$collect()
+
+    # Cast by column type (numeric/bool -> float), (other -> string)
+    # This is done later in py-polars but we need to do it here. When we gather
+    # values in the list, we coerce the types, meaning that stats for date (for
+    # instance) are coerced to numeric. Casting those to string beforehand
+    # fixes this.
+    df_metrics_schema <- df_metrics$schema
+    for (i in seq_along(df_metrics_schema)) {
+      nm <- names(df_metrics_schema)[i]
+      dtype <- df_metrics_schema[[i]]
+      df_metrics_dtype <- if (dtype$is_numeric() || inherits(dtype, "polars_dtype_boolean")) {
```

Comment:
I still think it would be nice to have `is_boolean()` and friends because this is kind of weird. Problem for later

Reply:
I recall pola-rs/polars#19235

Reply:
Yes, too bad they don't want to implement those